packages/lesswrong/components/common/FlashMessages.tsx

```diff
@@ -21,7 +21,15 @@ export const MessageContextProvider = ({children}: {
   const [messages,setMessages] = useState<AnyBecauseTodo[]>([]);
   
   const flash = useCallback((message: AnyBecauseTodo) => {
-    setMessages([...messages, message]);
+    if (!messages.length) {
+      setMessages([message])
+    } else {
+      // Show the transition for clearing the old message, then pop up the new one
+      setMessages(messages.map((message: AnyBecauseTodo) => ({...message, hide:true})));
+      setTimeout(() => {
+        setMessages([message]);
+      }, 500);
+    }
```

Comment:
@b0b3rt The previous behaviour was that if you did a new `flash` when there was already an existing message up it would never show the new message (because it only ever shows messages[0]). I'm assuming that was not the behaviour we wanted and this is better in all cases, but maybe there is a case I missed where the later elements in the messages array do do something?

Reply:
This is no longer needed for this PR, but I think it's a pareto improvement so I'm keeping it in